catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/util/MockCatroidApplication.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.util;
+
+import org.catrobat.catroid.CatroidApplication;
+import org.catrobat.catroid.uiespresso.web.util.DaggerMockWebComponent;
+import org.catrobat.catroid.uiespresso.web.util.MockWebRequestModule;
+
+public class MockCatroidApplication extends CatroidApplication {
```

Comment:
this is not a mock per definition (since its not replacing the CatroidApplication, its still there, fully blown, just the web component has been replaced), can you please find a more suitable name for this. this is a decorator, the only mock is the requestModule

Reply:
ok, ive seen now that multiple dagger tutorials use exactly this nomenclature, althouth im a fan of naming-things-what-they-are, if that is what is used everywhere else im ok...